dask_cloudprovider/gcp/instances.py

```diff
@@ -87,6 +88,9 @@ def __init__(
         self.disk_type = disk_type or self.config.get("disk_type")
         self.ngpus = ngpus or self.config.get("ngpus")
         self.network = network or self.config.get("network")
+        self.network_projectid = (
+            network_projectid if network_projectid is not None else projectid
```

Comment:
This is useful when your VM networks do not "live" in the same project as your network. Typical scenario in GCP is e.g. [Shared VPC](https://cloud.google.com/vpc/docs/shared-vpc). The MR intends to make this a backwards compatible change by providing a new argument that defaults to the original behavior (network project === vm project) when `network_projectid` is unspecified.